### PR TITLE
[CI] Add inductor cpu accuracy test running on AVX2 runners

### PR DESCRIPTION
**Pull request by zxd1997066**
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: .github/workflows/inductor.yml [hunk 1]
```diff
@@ -188,6 +188,13 @@ jobs:
           { config: "cpu_aot_inductor_torchbench_freezing", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
           { config: "cpu_aot_inductor_torchbench_freezing", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
           { config: "inductor_torchbench_cpu_smoketest_perf", shard: 1, num_shards: 1, runner: "linux.24xl.spr-metal" },
+          { config: "inductor", shard: 1, num_shards: 2, runner: "linux.10xlarge.avx2" },
```
Comment (desertfire) [inline .github/workflows/inductor.yml:191]: Can you add avx2 in the config names to make them more distinguishable?

File: test/test_ops.py [hunk 2]
```diff
@@ -272,6 +273,7 @@ def test_numpy_ref(self, device, dtype, op):
             in ("signal_windows_exponential", "signal_windows_bartlett")
             and dtype == torch.float64
             and "cuda" in device
+            or "cpu" in device
```
Comment (jgong5) [inline test/test_ops.py:276]: Should we do this instead?
```suggestion
            and device in ["cuda", "cpu"]
```

File: benchmarks/dynamo/check_accuracy.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 flaky_models = {
     "yolov3",
     "gluon_inception_v3",
+    "detectron2_maskrcnn_r_101_c4",
```
Comment (jgong5) [inline benchmarks/dynamo/check_accuracy.py:14]: This only happens with AVX2 test?

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: .github/workflows/inductor.yml (hunk 1 above)
Comment (zxd1997066) [reply .github/workflows/inductor.yml]: Thanks for the review. Added avx2 in the config names.

File: test/test_ops.py (hunk 2 above)
Comment (zxd1997066) [reply test/test_ops.py]: Thanks for the suggestion, applied the changes.

File: benchmarks/dynamo/check_accuracy.py (hunk 3 above)
Comment (zxd1997066) [reply benchmarks/dynamo/check_accuracy.py:14]: detectron2_maskrcnn_r_101_c4 has different accuracy status with different config in CI:
1. When running with avx512, it passed with freezing https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/ci_expected_accuracy/cpu_inductor_torchbench_freezing_inference.csv#L22, but failed without freezing https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/ci_expected_accuracy/cpu_inductor_torchbench_inference.csv#L21
2.  When running with freezing, it passed with avx512 but failed with avx2.

And I cannot reproduce the accuracy failure of this model on my local avx2 machine. In CI, It has the same failure msg as detectron2_maskrcnn_r_50_c4 discussed in https://github.com/pytorch/pytorch/issues/127073, suppose this model also has a numerical issue and mark it as flaky.

**Review by desertfire — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

File: test/test_ops.py (hunk 2 above)
Comment (zxd1997066) [reply test/test_ops.py]: This change cannot skip this test on cuda side. https://hud.pytorch.org/pr/pytorch/pytorch/128682#27568460296. We would like to keep previous code.

**Pull request closed without merging**